syntax = "proto3";

package GrpcDcopPb;

service GrpcRequest
{
	rpc ProcessRequest(PbRequest)returns(stream PbResult){}
	rpc CommandLineResult(PbRequest)returns(stream PbResult){}
}

service Notify2NCS
{
	rpc Notify(PbResult)returns(Errcode){}
}

message PbRequest {
	repeated string CmdLine = 1;
}

message PbResult {
	uint32 Result = 1;
	uint32 Count = 2;

	repeated string Field = 3;

	message ColValue {
		int32 Type = 1;
		int32 iVal = 2;
		uint32 dwVal = 3;
		uint64 qwVal = 4;
		double dbVal = 5;
		string strVal = 6;
	}
	
	message Col {
		repeated ColValue ColVal = 1;
	}

	repeated Col Row = 4;
	GrpcMsgType MsgType = 5;
	string Title = 6;
	string strResult = 7;
	uint32 NeId = 8;
}

message Errcode
{
	uint32 dwErrcode = 1;
}

enum GrpcMsgType
{
	GRPCMSGTYPE_BCMD = 0;
	GRPCMSGTYPE_STRCMD = 1;
	GRPCMSGTYPE_NOTIFYEVENT = 2;
	GRPCMSGTYPE_REPORTALARM = 3;
}
